### PR TITLE
Fix github workflows

### PR DESCRIPTION
**Pull request by Latropos**
## Summary

1. **Fix paths**
   - Workflow files have been renamed, but the paths inside the workflows weren't updated
   - Due to this mismatch the workflow is not triggered once it is edited
4. **Fix caching**
   - Remove caching when building apps (it wasn't working well)

## Test plan
There are failing tests, but their fail in not depending on this PR
image
image

### CODE REVIEW

File: .github/workflows/example-android-build-check.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ on:
     branches:
       - main
     paths:
-      - .github/workflows/android-build.yml
+      - .github/workflows//example-android-build-check.yml
```
Comment (tjzel) [inline .github/workflows/example-android-build-check.yml:21]:
```suggestion
      - .github/workflows/example-android-build-check.yml
```

File: .github/workflows/example-ios-build-check.yml [hunk 2]
```diff
@@ -1,10 +1,10 @@
-name: Example iOS build check
+name: Example app, iOS build check (paper and fabric)
```
Comment (tjzel) [inline .github/workflows/example-ios-build-check.yml:1]: Please revert this change, I made the workflow names match their filename for convenience.

File: .github/workflows/example-ios-build-check.yml [hunk 3]
```diff
@@ -45,43 +45,15 @@ jobs:
       - name: Check out Git repository
         uses: actions/checkout@v4
 
-      # TODO: Rework caching in this CI for monorepo setup.
```
Comment (tjzel) [inline .github/workflows/example-ios-build-check.yml:48]: Let's keep such comment here instead:

`# TODO: Add caching for node_modules and artifacts that will work with monorepo setup.`

File: .github/workflows/example-tvos-build-check.yml [hunk 4]
```diff
@@ -42,51 +43,20 @@ jobs:
       - name: Install monorepo node dependencies
         run: yarn install --immutable
 
-      - name: Restore Reanimated node_modules from cache
```
Comment (tjzel) [inline .github/workflows/example-tvos-build-check.yml:45]: Let's keep such comment here instead:

`# TODO: Add caching for node_modules and artifacts that will work with monorepo setup.`

File: .github/workflows/example-android-build-check.yml [hunk 5]
```diff
@@ -47,19 +47,6 @@ jobs:
           distribution: 'zulu'
           java-version: 17
 
-      # TODO: Rework caching in this CI for monorepo setup.
```
Comment (tjzel) [inline .github/workflows/example-android-build-check.yml:50]: Let's keep such comment here instead:

`# TODO: Add caching for node_modules and artifacts that will work with monorepo setup.`

**Review by tjzel — CHANGES_REQUESTED** [review]

File: .github/workflows/example-android-build-check.yml [hunk 6]
```diff
@@ -47,19 +47,6 @@ jobs:
...
-          path: .
-          key: ${{ runner.os }}-node-modules-${{ hashFiles('yarn.lock') }}
-          restore-keys: ${{ runner.os }}-node-modules-
-      - name: Restore app node_modules from cache
-        uses: actions/cache@v3
-        with:
-          path: ${{ matrix.working-directory }}/node_modules
-          key: ${{ runner.os }}-node-modules-${{ matrix.working-directory }}-${{ hashFiles(format('{0}/yarn.lock', matrix.working-directory)) }}
-          restore-keys: ${{ runner.os }}-node-modules-${{ matrix.working-directory }}-
       - name: Install monorepo node dependencies
```
Comment (Latropos) [inline .github/workflows/example-android-build-check.yml:50]:
```suggestion
      # TODO: Add caching for node_modules and artifacts that will work with monorepo setup.
      - name: Install monorepo node dependencies
```
